vtysh/vtysh_config.c

```diff
@@ -616,8 +616,14 @@ static int vtysh_read_file(FILE *confp, bool dry_run)
 	vty->node = CONFIG_NODE;
 
 	vtysh_execute_no_pager("enable");
-	vtysh_execute_no_pager("conf term file-lock");
-	vty->vtysh_file_locked = true;
+	/*
+	 * When reading the config, we need to wait until the lock is acquired.
+	 * If we ignore the failure and continue without the lock, the config
+	 * will be fully ignored.
+	 */
+	do {
+		ret = vtysh_execute_no_pager("conf term file-lock");
+	} while (ret == CMD_WARNING_CONFIG_FAILED);
```

Comment:
Shouldn't there be a sleep here rather than tight loop?

Reply:
I'll add a sleep.

Reply:
Done.